### ISSUE TITLE
WCAG: Tab order is wrong in repeating group v2

### ISSUE THREAD

**Issue by lassopicasso** [Altinn/app-frontend-react#1039]
### Description of the bug

This is related to issue #908 where the overall issue was fixed, but with additional scenarios the focus would stop working as intended. This is related to how the HTML is constructed for repeating groups. 

The repeating group container consists of a table of rows. Each row represents a group. New groups is added above the "add new group"-button. Hence the tab order doesn't visit the new group, but continues its path below.

So, the current workaround for #908, is to shift the focus upwards to the first focusable element in the edit container of the new group - using useRef and useEffect.

However, if the reference group is open and we then also opening a repeating group, the focus would be on reference instead of repeating. The reason for this is that both groups are rendered (reference is re-rendered when repeating is opened). And since reference is further down in the HTML DOM it will result in useRef using reference group as ref when the logic of focus is executed in useEffect. Ref. Video 1.

A similar issue occurs if we navigate back and forth in the edit container of repeating group. The focus will stop working as intended because it is no longer focused on the top element. Ref. Video 2.

We could have added more conditional statements to deal with these scenarios before executing the logic to shift focus. However, this would not be easy to work with in the long run.

A possible solution (in the short run?) is to move the buttons up in the HTML DOM so that new groups added are below them, but visually displayed as they are above. Here I also refer to the navigation buttons in the edit container of a repeating group. This would make it easier to deal with issues on the WCAG tab order.

In the longer run we may consider investing in an overall restructuring of repeating groups (and also group container?). 

**Update from UU test by SSB**
When pressing "Lagre og åpne neste" focus is moved to the previous "rediger" button. Focus should be on first field in the group. And if you press "Lagre og lukk" focus should be on "rediger" on the same row.

Comment from Lasse
Maybe some of theese issues will be fixed if we decide to go trough with this issue: https://github.com/Altinn/app-frontend-react/issues/1630
Making it a modal and locking the tab order inside the modal should be easier to control focus than opening up all of the tables.

Video 1:

https://user-images.githubusercontent.com/74791975/227233659-c17a934e-d8e0-4ee8-81fa-75544686c15d.mov

Video 2:

https://user-images.githubusercontent.com/74791975/227228927-a1f0c0ed-b44e-4e4c-aa71-b4b2702a1e44.mp4

### Steps To Reproduce

1. Go to a form with repeating groups

Issue 1:
2. Open a "reference group" then open a "repeating group"
3. Focus is on the reference group.. 

Issue 2: 
2. Open a "reference group".
3. Navigate back and forth. Focus stops working as intended.

**Comment by Febakke** [comment]
Might be related to #1630

**Comment by adamhaeger** [comment]
Setting this to blocked until we have decided what approach to go for, ref this convo: https://digdir.slack.com/archives/C076GU2QHCG/p1727961304612199

**Comment by olavflar** [comment]
Når det er løst så ligger det en guide her for å oppdatere tilgjengelighetserklæringen: https://digdir.atlassian.net/wiki/x/CgDy-/
